suites/message/nested.go

```diff
@@ -57,13 +57,15 @@ func MessageTest_NestedSends(t *testing.T, factory state.Factories) {
 		defer td.Complete()
 
 		stage := prepareStage(td, acctDefaultBalance, multisigBalance)
+		to, _ := td.NewAccountActor(address.SECP256K1, big.Zero())
 		balanceBefore := td.GetBalance(stage.creator)
 
-		// Multisig sends back to the creator.
 		amtSent := abi.NewTokenAmount(1)
-		result := stage.sendOk(stage.creator, amtSent, builtin.MethodSend, nil, nonce)
+		result := stage.sendOk(to, amtSent, builtin.MethodSend, nil, nonce)
 
-		td.AssertActor(stage.creator, big.Sub(big.Add(balanceBefore, amtSent), result.Receipt.GasUsed.Big()), nonce+1)
+		td.AssertActorChange(stage.creator, balanceBefore, result.Msg.GasLimit, result.Msg.GasPrice, result.Msg.Value, result.Receipt, result.Msg.CallSeqNum+1)
```

Comment:
Here and it many of the others you can use `amtSent`, or `big.Zero()` when nothing is sent, instead of `result.Msg.Value`. It's more concise and stronger, enforcing that the message that was processed has the result the test case knows it should.

Reply:
Also use the existing `nonce+1` for the same reason: the test is trying to assert that it knows what the nonce is, but you've removed that and exposed it to bugs in the calculation of `Msg.CallSeqNum`.